core/src/main/scala/org/apache/spark/util/collection/OpenHashMap.scala

```diff
@@ -29,6 +29,16 @@ import scala.reflect.ClassTag
  * NOTE: when using numeric type as the value type, the user of this class should be careful to
  * distinguish between the 0/0.0/0L and non-exist value
  */
+private[spark] class CollationAwareHashMap[K: ClassTag, V: ClassTag, X](
+    initialCapacity: Int,
+    hashering: AnyRef => Long,
+    equalsFunction: (AnyRef, AnyRef) => Boolean
+)
+  extends OpenHashMap[K, V](initialCapacity) {
+  override def getOpenHashSet: OpenHashSet[K] =
+    new CollationAwareOpenHashSet[K, X](initialCapacity, 0.7, hashering, equalsFunction)
```

Comment:
what's 0.7 here? since I see it in multiple lines, we should consider separating it out into one place, like a constant